extension/storage/README.md

```diff
@@ -0,0 +1,39 @@
+# Storage
+
+A storage extension persists state beyond the collector process. Other components can request a storage client from the storage extension and use it to manage state.
```

Comment:
how does one configure this extension? It seems to have no factory / configuration like normal extensions do.

Reply:
This package is intended only to define the API for storage extensions. There are a couple concrete implementations in contrib which each have their own factory, config, etc. The `file_storage` extension is used quite frequently: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/storage/filestorage

Reply:
I suggest making this explicit in the readme and in the code. Right now it looks nothing like an extension. Maybe even rename `storageapi` or have some common directory structure for APIs

Reply:
I've added a note to the top of the readme and verified that we have code comments to this effect.

I think the package naming is consistent with for example the `extension/auth` package, which serves a similar purpose.

Reply:
thanks.